File: internal/model/cargo.go
package model

import (
	"fmt"
	"time"
)

type Cargo struct {
	IsOpen         bool
	QRCode         string
	BottomDistance uint16
	UpdatedAt      time.Time
}

type CargoDoorMotor struct {
	Direction CargoDoorMotorDirection
	Speed     uint8
	IsRunning bool
	Enabled   bool
	UpdatedAt time.Time
}

type CargoDoorMotorDirection uint8

func (s CargoDoorMotorDirection) Validate() error {
	switch s {
	case CargoDoorDirectionClose, CargoDoorDirectionOpen:
		return nil
	default:
		return fmt.Errorf("invalid cargo door direction: %d", s)
	}
}

func (s CargoDoorMotorDirection) String() string {
	return []string{"CLOSE", "OPEN"}[s]
}

const (
	CargoDoorDirectionClose CargoDoorMotorDirection = iota
	CargoDoorDirectionOpen
)

File: ui/src/types/cargo.ts
export interface Cargo {
  isOpen: boolean
  qrCode: string
  bottomDistance: number
  hasItem: boolean
  updatedAt: string
}

export interface CargoDoorMotorState {
  direction: 'CLOSE' | 'OPEN'
  speed: number
  isRunning: boolean
  enabled: boolean
  updatedAt: string
}
